packages/payment-widget/src/lib/utils/request.ts

```diff
@@ -0,0 +1,238 @@
+import {
+  RequestNetwork,
+  Types,
+  Utils,
+} from "@requestnetwork/request-client.js";
+import { Web3SignatureProvider } from "@requestnetwork/web3-signature";
+import type { Currency } from "../types";
+import { Contract, providers, utils } from "ethers";
+import {
+  hasSufficientFunds,
+  approveErc20,
+  payRequest,
+} from "@requestnetwork/payment-processor";
+import { chains } from "./chains";
+
+export const prepareRequestParameters = ({
+  currency,
+  sellerAddress,
+  payerAddress,
+  amountInCrypto,
+  exchangeRate,
+  amountInUSD,
+}: {
+  currency: Currency;
+  sellerAddress: string;
+  payerAddress: string;
+  amountInCrypto: number;
+  exchangeRate: number;
+  amountInUSD: number;
+}) => {
+  const isERC20 = currency.type === Types.RequestLogic.CURRENCY.ERC20;
+  const currencyValue = isERC20 ? currency.address : "eth";
+
+  return {
+    requestInfo: {
+      currency: {
+        type: currency.type,
+        value: currencyValue,
+        network: currency.network,
+      },
+      expectedAmount: utils
+        .parseUnits(amountInCrypto.toString(), currency.decimals)
+        .toString(),
+      payee: {
+        type: Types.Identity.TYPE.ETHEREUM_ADDRESS,
+        value: sellerAddress,
+      },
+      payer: {
+        type: Types.Identity.TYPE.ETHEREUM_ADDRESS,
+        value: payerAddress,
+      },
+      timestamp: Utils.getCurrentTimestampInSecond(),
+    },
+    paymentNetwork: {
+      id: isERC20
+        ? Types.Extension.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT
+        : Types.Extension.PAYMENT_NETWORK_ID.ETH_FEE_PROXY_CONTRACT,
+      parameters: {
+        paymentNetworkName: currency.network,
+        paymentAddress: sellerAddress,
+        feeAddress: "0x0000000000000000000000000000000000000000",
+        feeAmount: "0",
```

Comment:
In V2, we may want to expose the feeAddress and feeAmount so that the platform can take a fee. FYI @Fpinto29